pkg/gatherers/common/gather_logs.go

```diff
@@ -80,6 +81,16 @@ func CollectLogsFromContainers( //nolint:gocyclo
 	var records []record.Record
 
 	for i := range pods.Items {
+		if len(containersFilter.PodNameRegexFilter) > 0 {
+			match, err := regexp.MatchString(containersFilter.PodNameRegexFilter, pods.Items[i].Name)
```

Comment:
It would be better to compile the regex outside of the loop.

Reply:
I would love to, but I can't, since it depends on `pods.Items[i].Name`.

Reply:
Well you can do `r,_ := regexp.Compile(resourceFilter.PodNameRegexFilter)` out side the for loop and then use the `r`